### PR TITLE
sheetify: rework internals, remove --out

### PR DESCRIPTION
**Pull request by yoshuawuyts**
- fixes npm package bug (`sf('my-package')` without assigning to var would bug out)
- fixes relative path detection (`./` was optional, but this was inconsistent with Node's `require()` api)
- removes `-o / --out` - shifting to a `css-extract` approach as a seperate plugin; greatly reduces internal complexity
- cleaned up internals so `transform.js` can actually be understood, lol
- semver: major

---

- [x] create `css-extract` plugin
- [x] update docs
- [x] update tests

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
`css-extract` is now live! https://github.com/stackcss/css-extract/tree/master

**Comment by yoshuawuyts** [comment]
Rad, tests updated and passing! `v5` is super close now :tada: @hughsk @ahdinosaur can you review? Thanks! 🍉

File: transform.js [hunk 1]
```diff
@@ -62,119 +64,93 @@ function transform (filename, options) {
...
+        if (err) return done(err)
+        const str = [
+          "((require('insert-css')(" + JSON.stringify(css) + ')',
+          ' || true) && ' + JSON.stringify(prefix) + ')'
+        ].join('')
+
+        const lolSemicolon = (val.node.parent.type === 'VariableDeclarator')
+          ? ''
+          : ';'
+        val.node.update(lolSemicolon + str)
```
Comment (ahdinosaur) [inline transform.js]: if we're not using `static-module`, we could probably go back to the way we used to transform where we add a new node before the `sheetify()` node with the `insert-css` call and update the `sheetify()` node with the prefix return value. or this is fine as well. :)

**Comment by ahdinosaur** [comment]
looks good to me, really appreciate your hard work @yoshuawuyts. :smiley_cat:

File: transform.js (hunk 1 above)
Comment (yoshuawuyts) [inline transform.js]: Hmm, yeah that would definitely result in more readable code. I think it's reasonable. I'm going to see if I can add it in 30 minutes, and else I'll leave this code as-is and someone else can add it, is that ok? In any case the `lolsemicolon` statement would have to remain, lest conflicts might occur.

**Comment by yoshuawuyts** [comment]
@ahdinosaur ^_____^ thanks man! Glad you're liking it! Not going to take all the credit for this tho, @hughsk and I worked hard last weekend to get this working - I just happened to be the one typing it out :sparkles:

File: transform.js (hunk 1 above)
Comment (yoshuawuyts) [inline transform.js]: I literally don't have the mental capacity to get this working :(

@ahdinosaur if you want to give it a go, perhaps as a different patch?

Comment (ahdinosaur) [inline transform.js]: i don't have capacity at the moment either, more than happy to postpone this for later when we feel the urge to change it.

**Comment by yoshuawuyts** [comment]
Cool! Given that @hughsk helped write everything but the tests, I think we should be all good! :D `sheetify@5`, here we go!

**Comment by yoshuawuyts** [comment]
released as `v5.0.0` :tada:

**Pull request closed without merging**